libcxx/test/std/ranges/range.adaptors/range.stride.view/iterator/operator.pass.cpp

```diff
@@ -0,0 +1,344 @@
+
+static_assert(is_relationally_equalable<std::ranges::iterator_t<EqualableView>>);
+static_assert(is_relationally_equalable<EqualableViewStrideViewIter>);
+
+static_assert(!std::three_way_comparable<std::ranges::iterator_t<EqualableView>>);
+static_assert(!std::ranges::random_access_range<EqualableView>);
+static_assert(!is_three_way_comparable<EqualableView>);
+
+using ThreeWayComparableView           = BasicTestView<rvalue_iterator<int*>>;
+using ThreeWayComparableViewStrideView = std::ranges::stride_view<ThreeWayComparableView>;
+using ThreeWayComparableStrideViewIter = std::ranges::iterator_t<ThreeWayComparableViewStrideView>;
+
+static_assert(std::three_way_comparable<std::ranges::iterator_t<ThreeWayComparableView>>);
+static_assert(std::ranges::random_access_range<ThreeWayComparableView>);
+static_assert(is_three_way_comparable<ThreeWayComparableStrideViewIter>);
+
+using UnEqualableView =
+    ViewOverNonCopyableIterator<cpp20_input_iterator<int*>, sentinel_wrapper<cpp20_input_iterator<int*>>>;
+using UnEqualableViewStrideView     = std::ranges::stride_view<UnEqualableView>;
+using UnEqualableViewStrideViewIter = std::ranges::iterator_t<UnEqualableViewStrideView>;
+
+static_assert(!is_relationally_equalable<std::ranges::iterator_t<UnEqualableView>>);
+static_assert(!is_relationally_equalable<UnEqualableViewStrideViewIter>);
+
+static_assert(!std::three_way_comparable<std::ranges::iterator_t<UnEqualableView>>);
+static_assert(!std::ranges::random_access_range<UnEqualableView>);
+static_assert(!is_three_way_comparable<UnEqualableView>);
+
+constexpr bool test_non_forward_operator_minus() {
+  using Base = BasicTestView<SizedInputIterator, SizedInputIterator>;
+  // Test the non-forward-range operator- between two iterators (i.e., ceil).
+  int arr[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  // First, what operators are valid for an iterator derived from a stride view
+  // over a sized input view.
+  using StrideViewIterator = std::ranges::iterator_t<std::ranges::stride_view<Base>>;
+
+  static_assert(is_plus_plusable_post<StrideViewIterator>);
+  static_assert(is_plus_plusable_pre<StrideViewIterator>);
+  static_assert(!is_minus_minusable_post<StrideViewIterator>);
+  static_assert(!is_minus_minusable_pre<StrideViewIterator>);
+  static_assert(!is_plus_equalable<StrideViewIterator>);
+  static_assert(!is_minus_equalable<StrideViewIterator>);
+  static_assert(!is_difference_plusable<StrideViewIterator>);
+  static_assert(!is_difference_minusable<StrideViewIterator>);
+  static_assert(!is_relationally_comparable<StrideViewIterator>);
+  static_assert(!is_indexable<StrideViewIterator>);
+
+  auto rav_zero    = Base(SizedInputIterator(arr), SizedInputIterator(arr + 10));
+  auto rav_one     = Base(SizedInputIterator(arr + 1), SizedInputIterator(arr + 10));
+  auto stride_zoff = std::ranges::stride_view(rav_zero, 3);
+  auto stride_ooff = std::ranges::stride_view(rav_one, 3);
```

Comment:
Please rename these: `zoff` and `ooff` are easy to mix up, and aren't really communicating enough information. I get `z` corresponds to `rav_zero` and `o` to `rav_one`, but I'm not sure about `off`.

Reply:
I believe that this is addressed in 10809679a823d8286b7bd45d3dbf9c23b2e4c4d5.